grug/app/src/app.rs

```diff
@@ -719,36 +672,26 @@ where
     VM: Vm + Clone,
     AppError: From<VM::Error>,
 {
+    let gas_tracker = ctx.replace_gas_tracker(GasTracker::new_limitless());
```

Comment:
While we let users pay for `withhold_fee`, we can't let user pay for `finalize_fee`.

Because taxman uses the gas consumption of everything up to before `finalize_fee` to calculate how much fee to charge.